component/src/main/java/io/siddhi/extension/io/cdc/source/polling/strategies/DefaultPollingStrategy.java

```diff
@@ -139,9 +155,14 @@ public void printEvent(Connection connection) {
                 handleEvent(detailsMap);
             }
         } catch (SQLException ex) {
+            if (metrics != null) {
+                isError = true;
```

Comment:
Shouldn't we set the isError variable outside the metrics null check?

Reply:
`isError` is only needed when metrics service is enabled, That's why I'm setting it inside the if the condition

Reply:
ack